### PR TITLE
Use divideAt in PointTool.addPoint so that adding new points doesn't affect curves' shape

### LINKED ISSUE #998: Adding a point to a curved segment with the reshape tool affects the segment's shape
### Expected Behavior

When you use the reshape tool to add a point in the middle of a curved segment, it should not affect that segment's shape. This is how Inkscape, for instance, behaves:
Peek 2020-04-11 11-05

### Actual Behavior

Adding new points to segments affects the shape of those segments:
Peek 2020-04-11 11-07

### Steps to Reproduce

1. Switch to the reshape tool
2. Select a shape that contains curved segments
3. Click along one of those curved segments to add a point
4. Observe how the segment's shape changes

### Operating System and Browser

All

### PR DESCRIPTION
**Pull request by adroitwhiz**
### Resolves

Resolves #998

### Proposed Changes

This PR replaces the subdivision logic in `PointTool.addPoint` with Paper.js' `divideAt` method.

### Reason for Changes

Currently, when adding points using the reshape tool, the shape of the curve segment where the point was added changes, because of how the `addPoint` subdivision logic works.

Paper.js' `divideAt` method will instead add a new point in the middle of a curve segment while not changing that segment's shape.

### CODE REVIEW

**Review by fsih — COMMENTED** [review]
Awesome! I didn't realize this existed. Works way better!

It handles straight edges differently. The old code would add a curved point, whereas the new one adds a point without handles. I can see people preferring either way. I like drawing a polygonal outline with the line tool, then adding in the curves with the add point tool. Maybe we should add this question to the design meeting as well?

**Comment by fsih** [comment]
Let's make all added points curved for now (i.e. add handles to them), to keep the behavior similar to the current behavior.

**Comment by adroitwhiz** [comment]
@fsih I replaced the custom smoothing logic with Paper.js' built-in `smooth` for segments added in the middle of straight lines. Is that alright?

File: src/helper/selection-tools/point-tool.js [hunk 1]
```diff
@@ -68,51 +67,16 @@ class PointTool {
...
-            handleIn = handleIn.multiply(3 * handleOut.length / handleIn.length);
-        }
-        if (handleOut.length > 3 * handleIn.length) {
-            handleOut = handleOut.multiply(3 * handleIn.length / handleOut.length);
-        }
+        const newSegment = hitProperties.hitResult.item.divideAt(hitProperties.hitResult.location);
 
-        const beforeSegment = hitProperties.hitResult.item.segments[hitProperties.hitResult.location.index];
-        const afterSegment = hitProperties.hitResult.item.segments[hitProperties.hitResult.location.index + 1];
+        // If we're adding a point in the middle of a straight line, it won't be smooth by default, so smoothe it
```
Comment (fsih) [inline src/helper/selection-tools/point-tool.js]: Typo smooth

**Review by fsih — DISMISSED** [review]
LG! Might want to wait until the week after the paper merge goes in to merge this.

**Review by fsih — APPROVED** [review]
LG, thanks!

Let's wait on this one, too

**Pull request merged**